fdbserver/VersionedBTree.actor.cpp

```diff
@@ -5986,8 +6204,12 @@ class VersionedBTree {
 		// Root pointer size is limited because the pager commit header is limited to smallestPhysicalBlock in
 		// size.
 		while (records.size() > 1 ||
-		       records.front().getChildPage().size() > (BUGGIFY ? 1 : BTreeCommitHeader::maxRootPointerSize)) {
+		       records.front().getChildPage().size() > (BUGGIFY ? 1 : BTreeCommitHeader::maxRootPointerSize) ||
+		       records[0].key != dbBegin.key) {
 			CODE_PROBE(records.size() == 1, "Writing a new root because the current root pointer would be too large");
+			CODE_PROBE(records[0].key != dbBegin.key,
```

Comment:
Update the comment before this loop as well since the new check changes its purpose.